core/lib/services/writables.spec.ts

```diff
@@ -0,0 +1,47 @@
+import {writable} from '@amadeus-it-group/tansu';
+import type {SpyInstance} from 'vitest';
+import {beforeEach, describe, expect, test, vi} from 'vitest';
+import {writableWithDefault} from './stores';
+import {typeArray} from './writables';
+
+describe(`Writables service`, () => {
+	let consoleErrorSpy: SpyInstance<Parameters<typeof console.error>, ReturnType<typeof console.error>>;
+
+	beforeEach(() => {
+		consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+	});
+
+	test(`should support typeArray`, () => {
+		const config$ = writable([0]);
+		const store$ = writableWithDefault<number[]>([0], config$, typeArray);
+		// set the wrong value
+		store$.set(100 as any);
+		expect(consoleErrorSpy).toHaveBeenCalledWith('Not setting invalid value', 100);
+		// the store is not updated with the wrong value
+		expect(store$()).toStrictEqual([0]);
+
+		const testArray = [15, 20];
+		store$.set(testArray);
+
+		expect(store$()).toStrictEqual([15, 20]);
+
+		testArray[1] = 15;
+		store$.set(testArray);
+
+		expect(store$()).toStrictEqual([15, 15]);
+	});
+
+	test(`typeArray equal function should compare values`, () => {
+		expect(typeArray.equal([15, 15], [15, 15])).toBe(true);
+		expect(typeArray.equal([15, 15], [15, 10])).toBe(false);
+		const arr = [15, 15];
+		const arr2 = arr;
+		arr2[1] = 10;
+		expect(typeArray.equal(arr, arr2)).toBe(true);
+		expect(typeArray.equal([15], [15, 15])).toBe(false);
+		expect(typeArray.equal([15, 15], [15])).toBe(false);
+		expect(typeArray.equal(undefined, [15])).toBe(false);
```

Comment:
`undefined` is not part of the accepted types for this equal function.
You should do `undefined as any` if you still want to test this case.

Reply:
true, will type the method params